### PR TITLE
update of pr template v1

### PR DESCRIPTION
**Pull request by gasstan**
> **Note**
>This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).

# Author

- [ ] **Self-review** your own code in GitHub's web interface[^1]
- [ ] Add **automated tests** as appropriate
- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing)[^2] as appropriate
- [ ] Check the **code coverage**[^3] report for the automated tests
- [ ] Update **documentation** as appropriate (e.g [README.md](../blob/main/README.md), and [**Architecture.md**](../blob/main/docs/Architecture.md), etc.)
- [ ] **Run the app** and try the changes
- [ ] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^4]

> **Note**
> It is good practice to provide before and after UI  **screenshots** in the description of this PR. This is only applicable for changes that modify the UI.

# Reviewer

- [ ] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
- [ ] Perform an **ad hoc review**[^5]
- [ ] Review the **automated tests**
- [ ] Review the **manual tests**
- [ ] Review the **documentation**, [**README.md**](../blob/main/README.md), and [**Architecture.md**](../blob/main/docs/Architecture.md) as appropriate
- [ ] **Run the app** and try the changes[^6]

[^1]: _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
[^2]: _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
[^3]: _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed. Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules._
[^4]: _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._
[^5]: _In addition to a first pass using the code review guidelines, do a second pass using your best judgement and experience which may identify additional questions or comments. Research shows that code review is most effective when done in multiple passes, where reviewers look for different things through each pass._
[^6]: _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._

### CODE REVIEW

**Comment by gasstan** [comment]
this is a first version of new pr template, feel free to add/remove/modify

**Comment by HonzaR** [comment]
Hi, thanks for providing us these 2 options. I like this simplification. I probably like this PR template a bit more, as it still keeps all the related information, as the original one does. I still believe we can benefit from having it here, for example, a fresh new contributor may be interested in reading it or scanning it by eyes, at least, before submitting his/her first PR and hopefully follow these rules, and thus save us some time. What do you think?

I have a few notes on it:
- The links to additional information below do not work. 3rd one is the only one working, can have all of them working?
- Just an idea, what about using a past form of the verbs here? Like: 
   - Self-reviewed your own code
   - Added appropriate automated tests if needed
   - Updated the as appropriate
   - etc.
- Still thinking about the new part of Note. I know that I don't need to say I'm providing screenshots, as it's obvious, but I also liked that as one of the check options. What is the main reason for moving those 2 check options to the Note section?

**Comment by HonzaR** [comment]
One more thing I like to discuss here is the first part of the template (This code review checklist is intended...). Can we somehow optically differentiate it? Because if a dev adds some additional text before the template text, which is desired, it always optically merges with the template text. What about dividing it by a line, or making it Note, or making it italic, or moving it below the list? What do you think?

**Comment by ccjernigan** [comment]
> One more thing I like to discuss here is the first part of the template (This code review checklist is intended...). Can we somehow optically differentiate it? Because if a dev adds some additional text before the template text, which is desired, it always optically merges with the template text. What about dividing it by a line, or making it Note, or making it italic, or moving it below the list? What do you think?

Making it a note seems like a good option to me.  I think it is better above, rather than below, the list.

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -1,23 +1,35 @@
+<!-- NOTE: Do not modify these when initially opening the pull request.  This is a checklist template that you tick off AFTER the pull request is created. -->
```
Comment (ccjernigan) [inline .github/pull_request_template.md:1]: I think this should be closer to the author checklist like it was before.  Thoughts?

Moved to the beginning, the phrase "Do not modify these" doesn't have context anymore.  The comment is just for the author, not the reviewer, and is primarily aimed at first time contributors who might try to edit the checklist markdown.

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,23 +1,35 @@
...
-- [ ] Run the app: Did you run the app and try the changes? _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._
\ No newline at end of file
+- [ ] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
+- [ ] Perform an **ad hoc review**[^adHoc]
+- [ ] Review the the **automated tests**
+- [ ] Review the **manual tests**
+- [ ] Review the **documentation**, [**README.md**](../blob/main/README.md), and [**Architecture.md**](../blob/main/docs/Architecture.md) as appropriate
+- [ ] Run the app and try the changes[^runTheApp]
+
+> **Note**
```
Comment (ccjernigan) [inline .github/pull_request_template.md:27]: I feel like this note could probably go away.  Thoughts?

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -1,23 +1,35 @@
...
+> We encourage you to compare coverage before and after changes and when possible, leaving it in a better place.
+
+[^selfReview]: _Code often looks different when reviewing the diff in a browser, making it easier to spot potential bugs._
+[^manTest]: _While we aim for automated testing of the application, some aspects require manual testing. If you had to manually test something during development of this pull request, write those steps down._
+[^coverage]:_While we are not looking for perfect coverage, the tool can point out potential cases that have been missed. **Code coverage** can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules._
```
Comment (ccjernigan) [inline .github/pull_request_template.md:32]: We can probably remove the bold around `**Code coverage**` now that this is under a footnote.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -1,23 +1,35 @@
...
+- [ ] **Self-review** your own code in GitHub's web interface[^selfReview]
+- [ ] Add appropriate **automated tests** if needed
+- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing)[^manTest] as appropriate
+- [ ] Check the **code coverage**[^coverage] report for the automated tests
+- [ ] Update **documentation** as appropriate? (e.g [README.md](../blob/main/README.md), etc.)
+- [ ] **Run the app** and try the changes?
+- [ ] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^pullSquash]
+
+> **Note**
+> It is good practice to provide before and after UI  **screenshots** in the description of this PR This is only applicable for changes that modify the UI.
```
Comment (ccjernigan) [inline .github/pull_request_template.md:16]: I'm not sure whether I have an opinion about this being a checkbox versus being just a note.

We haven't had a ton of UI changes in PRs recently, so as a checkbox it gets under utilized. However we do expect more UI changes in the future.

If we keep it as a note, the text could be updated to read: "For changes that modify the UI, it is good practice to provide before and after **screenshots** in the description of this PR."

**Comment by ccjernigan** [comment]
> * Just an idea, what about using a past form of the verbs here? Like:
>   
>   * Self-reviewed your own code
>   * Added appropriate automated tests if needed
>   * Updated the as appropriate
>   * etc.

Good comment here.

I think I prefer the present tense.  A contributor who isn't familiar with our PRs will likely not have all of these things in place initially.  Past tense makes it seem like this is necessary to complete them before opening a PR, but we don't really care if they create a PR and then do these checks afterwards as long as it happens before a reviewer is assigned.

**Comment by gasstan** [comment]
> > One more thing I like to discuss here is the first part of the template (This code review checklist is intended...). Can we somehow optically differentiate it? Because if a dev adds some additional text before the template text, which is desired, it always optically merges with the template text. What about dividing it by a line, or making it Note, or making it italic, or moving it below the list? What do you think?
> 
> Making it a note seems like a good option to me. I think it is better above, rather than below, the list.

- I agree, placing it above will make it easier to notice. 
- should we put something like this above the checklist?
```
<!-- NOTE: write any additional comments here -->
> **Note**
> 
```

File: .github/pull_request_template.md (hunk 4 above)
Comment (gasstan) [reply .github/pull_request_template.md:16]: I think it depends on how do we want to communicate it.  
If we leave it as a note it is more of a suggestion, but we really don't care if you do it or not.
If we change it to checkbox it is perceived more as a requirement (no mater if mandatory, or optional)
what do you think? @HonzaR @ccjernigan

**Comment by HonzaR** [comment]
> > > One more thing I like to discuss here is the first part of the template (This code review checklist is intended...). Can we somehow optically differentiate it? Because if a dev adds some additional text before the template text, which is desired, it always optically merges with the template text. What about dividing it by a line, or making it Note, or making it italic, or moving it below the list? What do you think?
> > 
> > 
> > Making it a note seems like a good option to me. I think it is better above, rather than below, the list.
> 
> * I agree, placing it above will make it easier to notice.
> * should we put something like this above the checklist?
> 
> ```
> 
> > **Note**
> > 
> ```

Yes, this could work. I'd just put one blank line under the comment, so I can write under it right away.

**Comment by gasstan** [comment]
Updated

File: .github/pull_request_template.md [hunk 5]
```diff
@@ -1,23 +1,33 @@
...
-- [ ] Code coverage: Did you check the code coverage report for the automated tests?  _While we are not looking for perfect coverage, the tool can point out potential cases that have been missed._ Code coverage can be generated with: `./gradlew check` for Kotlin modules and `./gradlew connectedCheck -PIS_ANDROID_INSTRUMENTATION_TEST_COVERAGE_ENABLED=true` for Android modules.
-- [ ] Documentation: Did you update documentation as appropriate? (e.g [README.md](../blob/main/README.md), etc.)
-- [ ] Run the app: Did you run the app and try the changes?
-- [ ] Screenshots: Did you provide before and after UI screenshots in the description of this pull request?  _This is only applicable for changes that modify the UI._
-- [ ] Rebase and squash: Did you pull in the latest changes from the main branch and squash your commits before assigning a reviewer? _Having your code up to date and squashed will make it easier for others to review. Use best judgement when squashing commits, as some changes (such as refactoring) might be easier to review as a separate commit._
 
+- [ ] **Self-review** your own code in GitHub's web interface[^selfReview]
```
Comment (HonzaR) [inline .github/pull_request_template.md:8]: None of the links to the bottom part still not working for me. The only one, which works for me is the 3rd one. Are they working for you?

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -1,23 +1,33 @@
...
+- [ ] Add appropriate **automated tests** if needed
+- [ ] Update the [**manual tests**](../blob/main/docs/testing/manual_testing) [^manTest] as appropriate
+- [ ] Check the **code coverage**[^coverage] report for the automated tests
+- [ ] Update **documentation** as appropriate? (e.g [README.md](../blob/main/README.md), etc.)
```
Comment (HonzaR) [inline .github/pull_request_template.md:12]: This one and the next one are the only ones, which have a question mark at the end. Should we unify it?

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -1,23 +1,33 @@
...
+- [ ] **Run the app** and try the changes?
+- [ ] Pull in the latest changes from the **main** branch and **squash** your commits before assigning a reviewer[^pullSquash]
+
+> **Note**
+> It is good practice to provide before and after UI  **screenshots** in the description of this PR This is only applicable for changes that modify the UI.
```
Comment (HonzaR) [inline .github/pull_request_template.md:17]: Missing period mark after `PR`.

File: .github/pull_request_template.md [hunk 8]
```diff
@@ -1,23 +1,33 @@
...
-- [ ] How is code coverage affected? _We encourage you to compare coverage before and after changes and when possible, leaving it in a better place._
-- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
-- [ ] Run the app: Did you run the app and try the changes? _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._
\ No newline at end of file
+- [ ] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
+- [ ] Perform an **ad hoc review**[^adHoc]
+- [ ] Review the the **automated tests**
+- [ ] Review the **manual tests**
+- [ ] Review the **documentation**, [**README.md**](../blob/main/README.md), and [**Architecture.md**](../blob/main/docs/Architecture.md) as appropriate
+- [ ] Run the app and try the changes[^runTheApp]
```
Comment (HonzaR) [inline .github/pull_request_template.md:26]: `Run the app` could also be bold here to have the same pattern across all checkboxes.

File: .github/pull_request_template.md [hunk 9]
```diff
@@ -1,23 +1,33 @@
...
-- [ ] Automated tests: Did you review the automated tests?
-- [ ] Manual tests: Did you review the manual tests?
-- [ ] How is code coverage affected? _We encourage you to compare coverage before and after changes and when possible, leaving it in a better place._
-- [ ] Documentation: Did you review Docs, [README.md](../blob/main/README.md), and [Architecture.md](../blob/main/docs/Architecture.md) as appropriate?
-- [ ] Run the app: Did you run the app and try the changes? _While the CI server runs the app to look for build failures or crashes, humans running the app are more likely to notice unexpected log messages, UI inconsistencies, or bad output data. Perform this step last, after verifying the code changes are safe to run locally._
\ No newline at end of file
+- [ ] Check the code with the [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md) **checklist**
+- [ ] Perform an **ad hoc review**[^adHoc]
+- [ ] Review the the **automated tests**
```
Comment (HonzaR) [inline .github/pull_request_template.md:23]: Doubled `the`.

File: .github/pull_request_template.md [hunk 10]
```diff
@@ -1,23 +1,33 @@
+<!-- NOTE: write any additional comments here -->
+
 This code review checklist is intended to serve as a starting point for the author and reviewer, although it may not be appropriate for all types of changes (e.g. fixing a spelling typo in documentation).  For more in-depth discussion of how we think about code review, please see [Code Review Guidelines](../blob/main/docs/CODE_REVIEW_GUIDELINES.md).
```
Comment (HonzaR) [inline .github/pull_request_template.md:3]: I've understood that you'd like to make this part a Note. What changed your mind?

**Review by HonzaR — CHANGES_REQUESTED** [review]
Thank you for updating the template. I left several mostly formatting comments, which should we consider.

File: .github/pull_request_template.md (hunk 10 above)
Comment (gasstan) [reply .github/pull_request_template.md:3]: It, wasn't my intention to change this to note. do yo prefer it that way?

Comment (HonzaR) [reply .github/pull_request_template.md:3]: I thought that we all agreed on that in [this](https://github.com/zcash/secant-android-wallet/pull/632#issuecomment-1323414512) discussion. And you pasted the code fragment, which seemed like a note too, to me. But it's up to you.

**Review by HonzaR — APPROVED** [review]

**Comment by LukasKorba** [comment]
Considered how much you've already discussed and covered I see no other things to sort out. I think the template looks fine and should work well in the real world. Great job btw.

**Review by LukasKorba — APPROVED** [review]
LGTM

**Review by ccjernigan — APPROVED** [review]
I made a minor tweak to the initial comment, removing the word "note" since that has special meaning in the markdown block right below it.

**Pull request merged**